source/javascripts/hooks/useFormattedYml.ts

```diff
@@ -1,52 +1,17 @@
-import { useEffect, useRef, useState } from 'react';
-
-import { AppConfig } from '@/models/AppConfig';
+import { useQuery } from '@tanstack/react-query';
 import BitriseYmlApi from '@/core/api/BitriseYmlApi';
-import useMonolithApiCallback from '@/hooks/api/useMonolithApiCallback';
-
-const identityParser = (result: string): any => result;
-
-const useFormattedYml = (appConfig: AppConfig): string => {
-  const [yml, setYml] = useState(typeof appConfig === 'string' ? appConfig : '');
-  const formatAppConfigRef = useRef<(options?: RequestInit) => void>();
-  const { failed, result, call } = useMonolithApiCallback<string>(
-    '/api/cli/format',
-    {
-      method: 'POST',
-      headers: {
-        Accept: 'application/x-yaml, application/json',
-      },
-    },
-    identityParser,
-  );
-
-  // NOTE: call function isn't referentially stable
-  useEffect(() => {
-    formatAppConfigRef.current = call;
+import { BitriseYml } from '../core/models/BitriseYml';
+
+const useFormattedYml = (appConfig: BitriseYml) => {
+  return useQuery({
```

Comment:
I don't recommend using `useQuery` for POST requests. In my opinion, `useMutation` is better for this case.